src/components/Api.js

```diff
@@ -0,0 +1,113 @@
+export default class Api {
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+  }
+
+  checkResponse(res) {
+    if (!res.ok) {
+      return Promise.reject(`Error: ${res.status}`);
+    } else {
+      return res.json();
+    }
+  }
+
+  //methods
+  getInitialCards() {
+    return fetch(`${this._baseUrl}/cards`, {
+      headers: this._headers,
+    }).then((res) => this.checkResponse(res));
+  }
+
+  getUserInfo() {
+    return fetch(`${this._baseUrl}/users/me`, {
+      headers: this._headers,
+    }).then((res) => this.checkResponse(res));
+  }
+
+  getUserAndCards() {
+    //called to get all info to display on page
+    return Promise.all([this.getUserInfo(), this.getInitialCards()]) //return Promise array of method calls
+      .then(([userInfo, cards]) => {
+        console.log([userInfo, cards]);
+        return { userInfo, cards }; //return object containing userInfo and Card Data
+      })
+      .catch((err) => {
```

Comment:
It’s better not to use `catch` in `Api` because you need to `catch` possible errors at the very **end** of any request, and this end is in `index.js`  after all blocks `then`. And in real projects blocks `catch` usually open error popups and so on.